src/Shortcodes/Shortcode.php

```diff
@@ -208,6 +210,118 @@ public static function addGiftVoucherForm($atts)
             return $html;
         }
 
+        /*
+        * Returns User workshops
+        */
+        public static function myWorkshops($atts)
+        {
+            extract(
+                shortcode_atts(
+                    array(
+                        'email' => '',
+                    ),
+                    $atts
+                )
+            );
+
+            // For now we kept the user email empty as this value should be
+            // filled by a dependent plugin or on the theme level using the
+            // filter: admwpp_user_email_workshops.
+            $email = apply_filters('admwpp_user_email_workshops', $email);
+
+            $workshops = self::getUserWorkshops($email);
+            $template = self::getTemplatePath('my-workshops');
+            ob_start();
+            include $template;
+            $html = ob_get_contents();
+            ob_end_clean();
+
+            return $html;
+        }
+
+        public static function getUserWorkshops($email)
+        {
+            if ($email) {
+                $gql = 'query contacts {
+                  contacts(filters: [{field: emailAddress, operation: eq, value: "' . $email . '"}]) {
+                    edges {
+                      node {
+                        learners {
+                          edges {
+                            node {
+                              id
+                              event {
+                                id
+                                code
+                                title
+                                bookedPlaces
+                                start
+                                reserved
+                                type
+                                location {
+                                  id
+                                  name
+                                }
+                                courseTemplate {
+                                  id
+                                  code
+                                  title
+                                }
+                              }
+                            }
+                          }
+                        }
+                        personalName {
+                          firstName
+                          lastName
+                        }
+                      }
+                    }
+                  }
+                }';
```

Comment:
can we add graphql queries in a separate file to prevent file getting huge

Reply:
I don't believe its needed at the moment.
The file `shortcodes.php` is only around 350 lines so it's not that huge.
Also keeping the graphql in the only method where it's used has its advantages.
Once the query is used in multiple places then its a good time to refactor and put it in a variable in a separate file.